tests/test_profile_data.py

```diff
@@ -176,3 +176,23 @@ def test_profile_using_spark(df_spark):
     )
 
     assert len(generate_raw_profile_dataset([["first_name", "blank"]], linker)) == 0
+
+
+def test_profile_null_columns(caplog):
+
+    df = pd.DataFrame(
+        [
+            {"unique_id": 1, "test_1": 1, "test_2": None},
+        ]
+    )
+
+    linker = DuckDBLinker(df)
+
+    linker.profile_columns(["test_1", "test_2"])
+
+    captured_logs = caplog.text
+
+    assert (
+        "Warning: No charts produced for test_2 as the column only contains null values."
```

Comment:
Make this a multi-line string to satisfy the linter (or add a [#noqa](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint_and_format.html#error-suppression) mark if you're feeling particularly lazy.